strawberryfields/ops.py

```diff
@@ -1814,30 +1816,24 @@ def __init__(self, V, r=None, decomp=True, tol=1e-6):
         self.x_disp = r[:self.ns]
         self.p_disp = r[self.ns:]
 
-        self.decomp = decomp
-
         if decomp:
             th, self.S = williamson(V, tol=tol)
             self.pure = np.abs(np.linalg.det(V) - (self.hbar/2)**(2*self.ns)) < tol
             self.nbar = np.diag(th)[:self.ns]/self.hbar - 0.5
+
         super().__init__([V, r])
 
+        self.decomp = decomp
+
         # FIXME merge() probably does not work for Gaussians if r is not zero?
 
     def _apply(self, reg, backend, **kwargs):
-        if self.decomp:
-            # explicity raise a NotImplementedError,
-            # so that the engine knows to perform a decomposition.
-            raise NotImplementedError
-
         p = _unwrap(self.p)
         s = sqrt(sf.hbar / 2)  # scaling factor, since the backend API call is hbar-independent
         backend.prepare_gaussian_state(p[1]/s, p[0]/(s*s), reg)
 
     def _decompose(self, reg):
         # pylint: disable=too-many-branches
-        if not self.decomp:
-            return None  # refuse to decompose
```

Comment:
Why was this here before?

Reply:
This was required to 'hack' around the previous try-except logic for decompositions. Now that that has been removed, it is no longer needed.